test/martha/martha_v3.test.js

```diff
@@ -141,155 +159,121 @@ test.serial('martha_v3 resolves a valid DOS-style url', async (t) => {
 // [0] https://ucsc-cgl.atlassian.net/browse/AZUL-702
 // [1] https://broadinstitute.slack.com/archives/G011ZUKHCUX/p1597694952108600
 test.serial('martha_v3 resolves a valid DRS-style url', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(sampleDosResponse);
-    getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
+    const bond = bondUrls('dcf-fence');
+    const dos = dosUrls('abc', '123');
+
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(dos.dataobjectsUrl, null).resolves(sampleDosResponse);
+
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://abc/123' } }), response);
+
     t.is(response.statusCode, 200);
     const result = response.send.lastCall.args[0];
+    t.deepEqual(result, sampleDosMarthaResult(googleSAKeyObject));
```

Comment:
I can't remember if there was a specific reason why we might need to `{ ...result }` here. I've been changing these as I go and the tests continue to pass. Is there something I'm not thinking of that could cause this to be wrong?

Reply:
Oh right, I found it: `FailureResponse`. See changes to `mockRequest` in this PR.